src/transformers/models/t5/modeling_t5.py

```diff
@@ -709,18 +705,15 @@ def forward(
         if do_cross_attention:
             # the actual query length is unknown for cross attention
             # if using past key value states. Need to inject it here
-            if present_key_value_state is not None:
-                query_length = present_key_value_state[0].shape[2]
-            else:
-                query_length = None
+            query_length = cache_position[0]
```

Comment:
it does ahaha, but the catch is that for static cache we need the `max_length` (aka length of keys after cache is added). So I added a check on cache type in `T5Attention` to use `max_cache_shape()` if we're in static cache setting

We could also simply rely on `key.shape[-2]` which is cleaner. I don't think we had a special purpose to get `key_length` like that instead of getting simply shape, lmk if there was a reason for that